### PR TITLE
Improve storage aggregation

### PR DESCRIPTION
**Pull request by jtpalmer**
## Description

Add use of last modified start date to storage realm aggregation.

## Motivation and Context

These changes allow aggregation of new storage data without aggregating all storage data.  The changes follow the same pattern as the cloud realm (see #841).

## Tests performed

Added new test to check the number of periods aggregated after ingesting data for a single day (after data for multiple days has already been ingested and aggregated).

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project as found in the **CONTRIBUTING** document.
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

File: open_xdmod/modules/xdmod/regression_tests/post_ingest_test.sh [hunk 1]
```diff
@@ -34,3 +34,19 @@ then
...
+last_modified_start_date=$(date +'%F %T')
+sudo -u xdmod xdmod-ingestor --datatype storage
+agg_output=$(mktemp --tmpdir storage-aggregation-XXXXXXXX)
+sudo -u xdmod xdmod-ingestor --aggregate=storage --last-modified-start-date "$last_modified_start_date" | tee $agg_output
+for unit in day month quarter year; do
+    if ! grep -q "unit: $unit, periods: 1," $agg_output; then
+        echo Did not aggregate 1 period of storage data for unit $unit
+        exit 1
+    fi
+done
```
Comment (chakrabortyr) [inline open_xdmod/modules/xdmod/regression_tests/post_ingest_test.sh]: More of a genuine question than a comment, are there any other edge cases we'd need to test for in the upgrade path?

Comment (jtpalmer) [reply open_xdmod/modules/xdmod/regression_tests/post_ingest_test.sh]: There are probably edge cases we aren't testing, but this test is just intended to check one specific case.
